ui/src/public/track.ts

```diff
@@ -56,6 +56,15 @@ export interface TrackManager {
    * power to the user compared to e.g. purely filtering by name.
    */
   registerTrackFilterCriteria(filter: TrackFilterCriteria): void;
+
+  /**
+   * Register a timeline overlay renderer.
+   *
+   * Overlays are rendered on top of all tracks in the timeline view and can be
+   * used to draw annotations that span multiple tracks, such as flow arrows or
+   * vertical lines marking specific events.
+   */
+  registerTimelineOverlay(overlay: TimelineOverlay): void;
```

Comment:
discussed offline: maybe just registerOverlay. Imho TimelineOverline makes me thing too much of "you can only overlay on the topi bit" ratehr than the whole canvas.

Reply:
Agreed & done